### PR TITLE
Feat/custom header

### PR DESCRIPTION
**Pull request by dawnseeker8**
## PR description
Allow user to create a chain with custom header attached to HTTP RPC call. 

## Fixed Issue(s)
User can enable app secret in their infura project alone with API key, however, user is unable to attach the app secret into header in RPC call to infura in current version of orchestrate. this change will allow customer to attach any custom header defined in chain to node provider. 

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://github.com/consensys/orchestrate/blob/master/CHANGELOG.md).

### CODE REVIEW

File: pkg/toolkit/app/http/config/dynamic/service.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func (p *ReverseProxy) SetDefaults() {
 	}
 
 	if p.PassHostHeader == nil {
-		p.PassHostHeader = utils.Bool(true)
+		p.PassHostHeader = utils.Bool(false)
```
Comment (ggarri) [inline pkg/toolkit/app/http/config/dynamic/service.go:45]: why are you making PassHostHeader false by default? I don't remember if this change was required

File: scripts/deps/docker-compose.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ x-postgres-common: &postgres-common
 services:
   zookeeper:
     <<: *container-common
-    image: confluentinc/cp-zookeeper:5.4.3
+    image: confluentinc/cp-zookeeper:7.2.5.arm64
```
Comment (ggarri) [inline scripts/deps/docker-compose.yml]: is this change intended? you are making docker compose arm64 only compatible. IMO you could create a arm version of this docker compose with a name `docker-compose.arm.yml`

File: pkg/toolkit/app/http/config/dynamic/service.go (hunk 1 above)
Comment (dawnseeker8) [reply pkg/toolkit/app/http/config/dynamic/service.go:45]: this is transfer from your private repo, https://github.com/Consensys/orchestrate-priv/commit/4ebf9a4e#diff-d0166d7f02c38d4bce9000d99320183184966dc6c87f0df42b3b383729ad7737R12 
please refer to pkg/toolkit/app/http/config/dynamic/service.go

File: scripts/deps/docker-compose.yml (hunk 2 above)
Comment (dawnseeker8) [reply scripts/deps/docker-compose.yml]: Sorry, i mistakenly commit this from my mac, i will revert it back to original intel chip version.

File: scripts/deps/docker-compose.yml [hunk 3]
```diff
@@ -19,7 +19,7 @@ x-postgres-common: &postgres-common
 services:
   zookeeper:
     <<: *container-common
-    image: confluentinc/cp-zookeeper:5.4.3
+    image: confluentinc/cp-zookeeper:7.2.5
```
Comment (dawnseeker8) [inline scripts/deps/docker-compose.yml:22]: i have tested to use this version of zookeeper and confirm it is working ok.

File: scripts/deps/docker-compose.yml [hunk 4]
```diff
@@ -28,7 +28,7 @@ services:
 
   kafka:
     <<: *container-common
-    image: confluentinc/cp-kafka:5.4.3
+    image: confluentinc/cp-kafka:7.2.5
```
Comment (dawnseeker8) [inline scripts/deps/docker-compose.yml:31]: same here, test with 7.2.5 kafka, working ok.

File: scripts/deps/docker-compose.yml [hunk 5]
```diff
@@ -88,11 +100,11 @@ services:
     volumes:
       - db-data-api:/var/lib/postgresql/data
     ports:
-      - 5432
+      - 5432:5432
```
Comment (dawnseeker8) [inline scripts/deps/docker-compose.yml:103]: This port open for host access database to check whether data correct or not.

File: scripts/deps/docker-compose.yml [hunk 6]
```diff
@@ -88,11 +100,11 @@ services:
...
 
   vault:
     <<: *container-common
-    image: consensys/quorum-hashicorp-vault-plugin:v1.1.4
+    image: consensys/quorum-hashicorp-vault-plugin:v1.1.5
```
Comment (dawnseeker8) [inline scripts/deps/docker-compose.yml:107]: same here to upgrade version to fix a small bug.

File: services/api/integration-tests/proxy.go [hunk 7]
```diff
@@ -42,4 +46,37 @@ func (s *proxyTestSuite) TestProxy() {
 
 		require.NoError(t, err)
 	})
+
+	s.T().Run("should register chain connected to a protected rpc npc and create proxy to the node", func(t *testing.T) {
```
Comment (dawnseeker8) [inline services/api/integration-tests/proxy.go:50]: This is copied from existing commit to private repo. https://github.com/Consensys/orchestrate-priv/commit/4ebf9a4e#diff-cbbc585e82a59d16bf5cd7be95fa6862ff405acec4899d28b97aa74288819b09R50

**Review by ggarri — APPROVED** [review]
